apps/server/src/routes/apiRouter.ts

```diff
@@ -14,9 +14,19 @@ router.get('/', (_req, res) => {
 
 // any GET request in /api is sent to the integration controller
 router.get('/*', (req, res) => {
-  const action = req.path.substring(1);
+  let action = req.path.substring(1);
+  const actionArray = action.split('/');
+  
   const params = { payload: req.query };
-
+  
+  //TODO: this can be better
+  if (actionArray.length > 1) {
+    action = actionArray.shift();
+    params.payload = actionArray.reduceRight(
+      (parm, key, index) => (index === actionArray.length - 1 ? { [key]: params.payload } : { [key]: parm }),
+      {},
+    );
+  }
```

Comment:
todo: if we are doing it this way, I would suggest extracting into a function like `actionsFromParams` and unit testing it, just to make sure we cover the major caveats.

I also think this can be simplified like
```ts
if (actionArray.length > 1) {
  action = actionArray.shift();
  params.payload = actionArray.reverse().reduce((result, key) => ({ [key]: result }), params.payload);
}
```

Reply:
nice